aea/test_tools/test_cases.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains test case classes based on pytest for AEA end-to-end testing."""
+
+import os
+import shutil
+import subprocess  # nosec
+import sys
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from aea.cli import cli
+from aea.configurations.base import PublicId
+from aea.mail.base import Envelope
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.test_tools.exceptions import AEATestingException
+
+from tests.common.click_testing import CliRunner
+from tests.conftest import AUTHOR, CLI_LOG_OPTION
+
+
+class AEATestCase:
+    """Test case for AEA end-to-end tests."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the test class."""
+        cls.runner = CliRunner()
+        cls.cwd = os.getcwd()
+        cls.t = tempfile.mkdtemp()
+
+        # add packages folder
+        packages_src = os.path.join(cls.cwd, "packages")
+        packages_dst = os.path.join(cls.t, "packages")
+        shutil.copytree(packages_src, packages_dst)
+
+        os.chdir(cls.t)
+
+    @classmethod
+    def teardown_class(cls):
+        """Teardowm the test."""
+        os.chdir(cls.cwd)
+        try:
+            shutil.rmtree(cls.t)
+        except (OSError, IOError):
+            pass
+
+    def disable_ledger_tx(self, vendor_name, item_type, item_name):
+        """
+        Disable ledger tx by modifying item yaml settings.
+        Run from agent's directory and only for item with present strategy is_ledger_tx setting.
+
+        :param vendor_name: str vendor name.
+        :param item_type: str item type.
+        :param item_name: str item name.
+
+        :return: None
+        """
+        json_path = "vendor.{}.{}s.{}.models.strategy.args.is_ledger_tx".format(
```

Comment:
`models.strategy.args.is_ledger_tx` this command is way to specific. What you probably want is a method called `set_config(self, author, item_type, item_name, dotted_path, value):` which can set any value. Also note, you are missing typing specifiers!

Reply:
This was already needed in at least 2 different test cases. But I agree that we should have a wrapper for config updating.

Reply:
Ok but that will just inflate the test class. It is much cleaner and extensible if we have a generic wrapper method for the set_config rather than lots of the them for different config setting. As I mentioned a few times before, this test class should be as generic as possible.